src/object/srv_obj_migrate.c

```diff
@@ -968,10 +986,33 @@ migrate_fetch_update_bulk(struct migrate_one *mrone, daos_handle_t oh,
 		rc = rc ? : ret;
 	}
 
+	for (i = 0; rc == 0 && i < mrone->mo_iod_num; i++) {
+		if (mrone->mo_iods[i].iod_size == 0) {
+			/* Since rebuild will be done within the stable
+			 * epoch, so no other IO should attach these
+			 * data during rebuild, but if the container is
+			 * being destroyed at the same time, then it may
+			 * not pull anything from remote target, so let's
+			 * set rc and avoid vos_update_end(), otherwise it
+			 * may cause VOS segfault.
+			 */
+			D_DEBUG(DB_REBUILD,
+				DF_UOID" %p dkey "DF_KEY" "DF_KEY" nr %d/%d"
+				" eph "DF_U64"\n", DP_UOID(mrone->mo_oid),
+				mrone, DP_KEY(&mrone->mo_dkey),
+				DP_KEY(&mrone->mo_iods[i].iod_name),
+				mrone->mo_iod_num, i, mrone->mo_epoch);
+			rc1 = -DER_DATA_LOSS;
```

Comment:
Given that DER_DATA_LOSS error is eventually ignored in this function, I don't see why you set it here.

Reply:
set here, because vos_update_end will skip such iod with error code, then it will be reset to 0 later. 

It can be any specific error code actually, but I thought DATA_LOSS seems a bit proper here. no?